internal/indexer/indexer.go

```diff
@@ -854,6 +866,497 @@ func (i *Indexer) indexReferenceToExternalDefinition(p *packages.Package, docume
 	return rangeID, true
 }
 
+type def struct {
+	pkg   *packages.Package
+	obj   types.Object
+	ident *ast.Ident
```

Comment:
consider storing the method list here, so we only have to look it up once per type (we use it again later)